dbld/prepare-release

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+set -e
+
+cd /source
+VERSION=$1
+
+OLD_VERSION=`cat VERSION`
+OLD_MAJOR=$(echo $OLD_VERSION | cut -d. -f1-2)
+NEW_MAJOR=$(echo $VERSION | cut -d. -f1-2)
+USER_EMAIL=$(git config --get user.email || echo "noemail@set.any.where")
+USER_NAME=$(git config --get user.name || echo "unknown user")
+VERSION_FILE_LIST="README.md doc/man/*.xml
+	scl/syslog-ng.conf
+	contrib/openbsd-packaging/syslog-ng.conf
+	packaging/debian/syslog-ng.conf
+	packaging/rhel/syslog-ng.conf"
+
+function update_version() {
+	echo Updating VERSION file
+	echo $VERSION > VERSION
+}
+
+function update_packaging_debian() {
+	echo Updating Debian packaging
+	DEB_FULLNAME=$USER_NAME DEB_EMAIL=$USER_EMAIL debchange --distribution unstable --newversion "$VERSION-1" -c packaging/debian/changelog "New upstream release $VERSION"
+}
+
+function update_packaging_rhel() {
+	echo Updating RHEL packaging
+	sed -i -e "/^%changelog/ a * `date '+%a %b %e %Y'` $USER_NAME <$USER_EMAIL> - $VERSION-1\n- updated to $VERSION\n" packaging/rhel/syslog-ng.spec
+}
+
+function update_version_refs_in_source() {
+	echo Updating version references in source files
+	ESCAPED_OLD_MAJOR=$(echo $OLD_MAJOR | sed -e 's/[]\/$*.^[]/\\&/g') &&
+	ESCAPED_NEW_MAJOR=$(echo $NEW_MAJOR | sed -e 's/[\/&]/\\&/g') &&
+	git ls-files $VERSION_FILE_LIST | xargs sed -i "s/$ESCAPED_OLD_MAJOR/$ESCAPED_NEW_MAJOR/g"
```

Comment:
@bazsi : I opened #3368 - for addressing this note -  If that PR merged, then I'll modify the unit tests and test configs.
At the end we won't need that huge sed, we can reduce its scope.

Reply:
Please note that the tool didn't commit the patch unconditionally, rather it just presented it to the user performing the release. However, it is true that in the long term it would be great if prepare-release could become completely automated.

The "@version: current" idea is a good one, however I don't thing we should document it. User configuration should never use it as we would be losing the upgrade mechanism.

Reply:
Also, please note that the latest version of the patch didn't change lib/tests/test_lexer.c, the list of files to be changed are listed in this variable:

VERSION_FILE_LIST="README.md doc/man/*.xml
        scl/syslog-ng.conf
        contrib/openbsd-packaging/syslog-ng.conf
        packaging/debian/syslog-ng.conf
        packaging/rhel/syslog-ng.conf"

Reply:
I thought of an alternative of the `sed` in cases, where `version: current` cannot be used.

We could create a `.in` file for every file, the config version is used. The `.in` file would have some kind of templating, like `@@CONFIG_VERSION@@`, which would be resolved in its `Makefile.am`. It would be enough to just create a list, which gathers files like these, and add each file to that list. The list would then be processed in the main `Makefile.am`. In the end it would be a `sed`, yes, but it would be much cleaner. Of course the same should be done in cmake.

If you agree with this, I am happy to implement it.